dao/src/test/scala/za/co/absa/enceladus/dao/rest/JsonSerializerSuite.scala

```diff
@@ -143,14 +187,68 @@ class JsonSerializerSuite extends BaseTestSuite with VersionedModelMatchers {
         |        "field": "",
         |        "oldValue": null,
         |        "newValue": null,
-        |        "message": "Test"
+        |        "message": "Dataset Test created."
         |      }
         |    ]
         |  }
         |}""".stripMargin
 
-      "deserializing should not throw" in {
-        JsonSerializer.fromJson[Dataset](datasetJson)
+      val dataset: Dataset = DatasetFactory.getDummyDataset(
+        name = "Test",
+        version = 5,
+        description = Some("some description here"),
+        hdfsPath = "/bigdata/test",
+        hdfsPublishPath = "/bigdata/test2",
+        schemaName = "Cobol1",
+        schemaVersion = 3,
+        dateCreated = ZonedDateTime.parse("2019-07-22T08:05:57.47Z"),
+        userCreated = "system",
+        lastUpdated = ZonedDateTime.parse("2020-04-02T15:53:02.947Z"),
+        userUpdated = "system",
+
+        conformance = List(
+          CastingConformanceRule(0,
+            outputColumn = "ConformedInt",
+            controlCheckpoint = false,
+            inputColumn = "STRING_VAL",
+            outputDataType = "integer"
+          ),
+          MappingConformanceRule(1,
```

Comment:
**Minor:** Could you add another `MappingConformanceRule` without the field of `mappingTableFilter`, to prove backward compatibility.

Reply:
Good idea, thanks, added.